src/parser.c

```diff
@@ -196,7 +202,10 @@ int ecs_parse_component_expr(
 
         if (prev_is_0) {
             /* 0 can only apppear by itself */
-            ecs_abort(ECS_INVALID_SIGNATURE, sig);
+            char err[50] = "/0";
+            build_error_string(err, sig, ptr - sig, system_id);
+
+            ecs_abort(ECS_INVALID_SIGNATURE, err);
```

Comment:
I think it's nicer to first print the error (in a format like I described above) and then call the abort. Otherwise the error message gets appended to the abort, and you get very long error messages which likely won't fit on a single line.

So something like:

```c
print_error_string(system_id, sig, "Unknown component '%s'", component_id);
ecs_abort(ECS_INVALID_SIGNATURE, NULL);
```

Reply:
Done. Still, the error and the printf are mixed together. Like the error starts to print the message while the ptintf is still working.